Content.Server/_White/Headcrab/HeadcrabSystem.cs

```diff
@@ -0,0 +1,202 @@
+        RemCompDeferred<PacifiedComponent>(uid);
+        var combatMode = EnsureComp<CombatModeComponent>(uid);
+        _combat.SetInCombatMode(uid, true, combatMode);
+        EnsureComp<NPCMeleeCombatComponent>(uid);
+        _npcFaction.RemoveFaction(args.Equipee, "Zombie");
+    }
+
+    private void OnMeleeHit(EntityUid uid, HeadcrabComponent component, MeleeHitEvent args)
+    {
+        if (!args.HitEntities.Any()
+            || _random.Next(1, 101) <= component.ChancePounce)
+            return;
+
+        TryEquipHeadcrab(uid, args.HitEntities.First(), component);
+    }
+
+    private void OnJump(EntityUid uid, HeadcrabComponent component, JumpActionEvent args)
+    {
+        if (args.Handled || component.EquippedOn is { Valid: true })
+            return;
+
+        args.Handled = true;
+        var xform = Transform(uid);
+        var mapCoords = _transform.ToMapCoordinates(args.Target);
+        var direction = mapCoords.Position - _transform.GetMapCoordinates(xform).Position;
+
+        _throwing.TryThrow(uid, direction, 7F, uid, 10F);
+        if (component.JumpSound != null)
+            _audioSystem.PlayPvs(component.JumpSound, uid, component.JumpSound.Params);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<HeadcrabComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            comp.Accumulator += frameTime;
+
+            if (comp.Accumulator <= comp.DamageFrequency)
+                continue;
+
+            comp.Accumulator = 0;
+
+            if (comp.EquippedOn is not { Valid: true } targetId ||
+                HasComp<ZombieComponent>(comp.EquippedOn) ||
+                _mobState.IsDead(uid))
+                continue;
+
+            if (!_mobState.IsAlive(targetId))
+            {
+                _inventory.TryUnequip(targetId, "mask", true, true);
+                comp.EquippedOn = EntityUid.Invalid;
+                continue;
+            }
+
+            _damageableSystem.TryChangeDamage(targetId, comp.Damage);
+            _popup.PopupEntity(Loc.GetString("headcrab-eat-entity-face"),
+                targetId, targetId, PopupType.LargeCaution);
+            _popup.PopupEntity(Loc.GetString("headcrab-eat-other-entity-face",
+                ("entity", targetId)), targetId, Filter.PvsExcept(targetId), true);
+        }
+    }
+
+        private bool TryEquipHeadcrab(EntityUid uid, EntityUid target, HeadcrabComponent component)
+    {
+        if (_mobState.IsDead(uid)
+            || !_mobState.IsAlive(target)
+            || !HasComp<HumanoidAppearanceComponent>(target)
+            || HasComp<ZombieComponent>(target))
+            return false;
+
+        _inventory.TryGetSlotEntity(target, "head", out var headItem);
+        return !HasComp<IngestionBlockerComponent>(headItem)
+            && !_inventory.TryEquip(target, uid, "mask", true);
```

Comment:
```suggestion
            && _inventory.TryEquip(target, uid, "mask", true);
```
Why is it returning false if we successfully equip the headcrab?